qcengine/programs/psi4.py

```diff
@@ -57,64 +57,113 @@ def compute(self, input_model: 'ResultInput', config: 'JobConfig') -> 'Result':
+                input_data["success"] = False
+                input_data["return_output"] = True
+
+                if input_data["schema_name"] == "qcschema_input":
+                    input_data["schema_name"] = "qc_schema_input"
+
+                # Execute the program
+                success, output = execute([which("psi4"), "--scratch", tmpdir, "--json", "data.json"],
+                                          {"data.json": json.dumps(input_data)}, ["data.json"],
+                                          scratch_directory=tmpdir)
+
+                if success:
+                    output_data = json.loads(output["outfiles"]["data.json"])
+                    if "extras" not in output_data:
+                        output_data["extras"] = {}
+
+                    # Check QCVars
+                    local_qcvars = output_data.pop("psi4:qcvars", None)
+                    if local_qcvars:
+                        # Edge case where we might already have qcvars, should not happen
+                        if "qcvars" in output_data["extras"]:
+                            output_data["extras"]["local_qcvars"] = local_qcvars
+                        else:
+                            output_data["extras"]["qcvars"] = local_qcvars
+
+                    if output_data["success"] is False:
+                        if "error_message" not in output_data["error"]:
+                            error_message = output_data["error"]
+                            error_type = "internal_error"
+                        else:
+                            error_message = output_data["error"]["error_message"]
+                            error_type = output_data["error"]["error_type"]
+
                     else:
-                        output_data["extras"]["qcvars"] = local_qcvars
+                        compute_success = True
+
+                else:
+                    error_message = output["stderr"]
+                    error_type = "execution_error"
+
+                # Reset the schema if required
+                output_data["schema_name"] = "qcschema_output"
+                output_data.pop("memory", None)
+                output_data.pop("nthreads", None)
+                output_data["stdout"] = output_data.pop("raw_output", None)
 
-                if output_data["success"] is False:
-                    if "error_message" not in output_data["error"]:
-                        # older c. 1.3 message-only run_json
-                        output_data["error"] = {"error_type": "internal_error", "error_message": output_data["error"]}
             else:
-                output_data = input_data
-                output_data["error"] = {"error_type": "execution_error", "error_message": output["stderr"]}
 
-        # Reset the schema if required
-        output_data["schema_name"] = "qcschema_output"
+                if ("psiapi" in input_model.extras) and input_model.extras["psiapi"]:
+                    import psi4
```

Comment:
I strongly expect that we'll want [this trio](https://github.com/psi4/psi4/blob/master/psi4/driver/schema_wrapper.py#L205-L207) here. ok to merge, and I'll find out for sure with DDD.

Reply:
This should happen Psi4 side IMO and not here.